File: c/isPalindromeNum.c
#include <stdio.h>
#include <stdbool.h>

bool isPalindromeNum(long long int x) {
    long long int tmp = x, result = 0;

    if (x < 10)
        return (x > 0) ? true : false;
    while (x != 0) {
        result = result * 10 + (x % 10);
        x /= 10;
    }

    return tmp == result ? true : false;
}

int main () {
    int x = 12321;
    bool result = isPalindromeNum(x);
    printf("%d", result);
    return 0;
}

File: zig/src/leetcode/isPalindromeNum.zig
pub fn isPalindromeNum(comptime T: type, x: T) bool {
    if (x < 10) return x >= 0;

    var n: T = x;
    var result: T = 0;

    while (n != 0) : (n = @divTrunc(n, 10))
        result = result * 10 + @rem(n, 10);

    return x == result;
}
